tests/common/ExecutionHelper.cs

```diff
@@ -231,8 +236,13 @@ public bool HasError (string prefix, int number, string message)
 
 		public void AssertWarningCount (int count, string message = "warnings")
 		{
-			if (count != WarningCount)
-				Assert.Fail ($"{message}\nExpected: {count}\nBut was: {WarningCount}\nWarnings:\n\t{string.Join ("\n\t", this.Messages.Where ((v) => v.IsWarning).Select ((v) => v.ToString ()))}");
+			AssertWarningCount (messages, count, message);
+		}
+
+		public static void AssertWarningCount (List<ToolMessage> messages, int count, string message = "warnings")
+		{
+			if (count != GetWarningCount (messages))
```

Comment:
Why not:
```
Assert.AreEqual (count, GetWarningCount (messages), \nWarnings:\n\t{string.Join ("\n\t", messages.Where ((v) => v.IsWarning).Select ((v) => v.ToString ()))}");
```

Reply:
Because that will create the string to report to the user in all cases, even when the assert doesn't fail, and there can be a significant number of messages in some cases.

In other words: it's a performance improvement.

Reply:
👍